lib/phoenix_token_auth/registrator.ex

```diff
@@ -7,15 +7,28 @@ defmodule PhoenixTokenAuth.Registrator do
   Returns a changeset setting email and hashed_password on a new user.
   Validates that email and password are present and that email is unique.
   """
+  def changeset(params = %{"username" => username}) when username != "" and username != nil do
+    Changeset.cast(struct(UserHelper.model), params, ~w(username))
+    |> Changeset.validate_change(:username, &Util.presence_validator/2)
+    |> Changeset.validate_unique(:username, on: Util.repo)
+    |> Changeset.put_change(:hashed_confirmation_token, nil)
+    |> Changeset.put_change(:confirmed_at, Ecto.DateTime.utc)
+    |> changeset_helper
+  end
+
   def changeset(params) do
     Changeset.cast(struct(UserHelper.model), params, ~w(email))
     |> Changeset.validate_change(:email, &Util.presence_validator/2)
     |> Changeset.validate_unique(:email, on: Util.repo)
+    |> changeset_helper
+  end
+
+  def changeset_helper(changeset) do
+    changeset
```

Comment:
was the problem with allowing both email and username that the uniqueness validation failed, if one of them is nil? if so, i think we should build a custom validator that only adds the error if the field is not nil. what do you think?